[v18] Edit the Resource Request guide (#52393)

ptgott opened the pull request:
* Edit the Resource Request guide

Closes #33417

Focus the guide more on its stated topic by moving tangentially related reference information at the end of the guide to more appropriate pages.

- Replace the `search_as_roles` section with a link. Much of the "Next steps" section is a long list of examples of Teleport roles that an admin could assign to `search_as_roles`. This is unnecessary, since a user familiar with a Teleport role won't need this information to understand `search_as_roles`. It is not the job of the Resource Requests guide to teach users about Teleport roles.
- Condense the "Next steps" guidance. There are "Next steps" sections. Make them one. Also turn the section re: Access Request plugins into a bullet.
- Create a "Request Access to Resources" guide that shows end-users how to create Access Requests. Instead of manually adding an entry to the User Guides section in `sidebar.json`, this change configures the sidebar to generate entries for this section automatically.
- Add a subsection to the Kubernetes RBAC guide explaining implications for Access Requests and Kubernetes cluster resources. This information was removed from the docs when we cleaned up the Resource Requests page.

* Respond to feedback

- Indicate that Teleport Enterprise is required to create Access Requests in the Web UI.
- Mention Teleport Connect.
- Mention `kube_resource`.

ptgott committed:
Edit the Resource Request guide (#52393)

* Edit the Resource Request guide

Closes #33417

Focus the guide more on its stated topic by moving tangentially related
reference information at the end of the guide to more appropriate pages.

- Replace the `search_as_roles` section with a link. Much of the "Next
  steps" section is a long list of examples of Teleport roles that an
  admin could assign to `search_as_roles`. This is unnecessary, since a
  user familiar with a Teleport role won't need this information to
  understand `search_as_roles`. It is not the job of the Resource Requests
  guide to teach users about Teleport roles.
- Condense the "Next steps" guidance. There are "Next steps" sections.
  Make them one. Also turn the section re: Access Request plugins into a
  bullet.
- Create a "Request Access to Resources" guide that shows end-users how
  to create Access Requests. Instead of manually adding an entry to the
  User Guides section in `sidebar.json`, this change configures the
  sidebar to generate entries for this section automatically.
- Add a subsection to the Kubernetes RBAC guide explaining implications
  for Access Requests and Kubernetes cluster resources. This information
  was removed from the docs when we cleaned up the Resource Requests
  page.

* Respond to feedback

- Indicate that Teleport Enterprise is required to create Access
  Requests in the Web UI.
- Mention Teleport Connect.
- Mention `kube_resource`.

r0mant reviewed (changes requested):
I'm not sure why we're moving access requests guide from under the Identity Governance section? In fact, I have the exact opposite goal for this quarter and I want to consolidate and reorganize everything related to IG under its corresponding docs section. Otherwise it's impossible to find anything.

If the issue is the resources requests guide is too big can we perhaps just split it up and add another guide under access requests section?

r0mant commented on docs/pages/connect-your-client/request-access.mdx:
````diff
@@ -0,0 +1,302 @@
+permissions for the cluster-scoped resource.  This is an example of the required permissions to
+request access only to the namespace `prod` (does not include any resources within it):
+
+```yaml
+kind: role
+version: v8
+spec:
+  allow:
+    kubernetes_resources:
+    - kind: namespaces
+      api_group: ''
+      name: prod
+      verbs:
+      - '*'
+```
````
If we don't want end users to see yaml, should we remove this?

ptgott replied:
Yes, I'll remove it.

r0mant commented on docs/pages/identity-governance/access-requests/resource-requests.mdx:
````diff
@@ -261,450 +252,15 @@ Getting updated certificates...
 iot:~ alice$
 ```
 
-### Restrict the resources a user can request access to
````
I can understand moving purely tsh related content to the clients section but it seems that this removes a big chunk of the documentation that explains how to configure access to various resources which is useful. Is this content available also elsewhere or is it just gone?

ptgott replied:
This section has ended up becoming a series of examples of Teleport roles with no relationship to Access Requests, making the page long and convoluted. We could preserve the section introduction (to add detail about `search_as_roles`) while removing the unnecessary H4 sections, linking to different access controls pages we already have in the docs.

ptgott replied:
On second thought, this section assumes that the guide shows the reader how to use `search_as_roles` to limit the resources a user can request access to, but there's actually no mention of `search_as_roles` before this point. The "Next steps" section includes a link to the Access Request configuration reference, which does explain `search_as_roles`. It might make sense to keep all `search_as_roles` content in the configuration guide as it is now.

r0mant commented on docs/pages/identity-governance/access-requests/resource-requests.mdx:
````diff
@@ -261,450 +252,15 @@ Getting updated certificates...
-    windows_desktop_logins: ["{{internal.windows_logins}}"]
-```
-
-### Request access to Kubernetes resources
````
I think it's find if the tsh related content lives under clients section but I do think that it's needed under main Identity Governance section as well, even if it's a link to the respective tsh guide. Otherwise I will be wondering how to actually use and test what I've just configured.

ptgott replied:
The how-to guide in Identity Governance includes basic `tsh` commands for testing out the resource request flow, but it does make sense to include a link to the user guide. I'll add that.

r0mant reviewed (commented):
I realize this is a backport but I do think we need to address the feedback before it makes it into the documentation.

ptgott committed:
Restore role examples in Resource Requests guide

- Remove YAML role example from the Access Request user guide, since the
  guide is not intended for admins who have access to Teleport dynamic
  resources.
- Add a link to the Access Request user guide in the Resource Requests
  page.

ptgott commented:
TODO:
- [ ] Once feedback is resolved, forward-port changes to `master`

ptgott commented:
@r0mant Checking if there is more to do on this one. Thanks!
